Sources/Swift/Protocol/SentryRedactOptions.swift

```diff
@@ -0,0 +1,7 @@
+import Foundation
+
+@objc
+protocol SentryRedactOptions {
```

Comment:
Decoupling.

If we continue along the current path, where everything depends on everything else, it will be impossible to make any major changes in the future.

I am trying to be a responsible engineer and adopting protocols-oriented programming, as suggested by Apple when Swift was released. 
I will create a document for us to discuss swift architecture since we're in the beginning. 
Also, this approach is highly beneficial for testing purposes. We no longer need to override a class with a multitude of behaviors; instead, we simply need to implement the protocol

Reply:
SentryViewPhotographer to have a dependency on sentryreplayoptions, 
SentryViewPhotographer will also be used for screenshot integration later.